spec/models/tool_spec.rb

```diff
@@ -23,5 +23,36 @@
       tool.should_not allow_value(-1).for(:total)
       tool.should allow_value(1).for(:total)
     end
+
+    it 'check rent with no free' do
+      tool = create(:tool, total: 1)
+      rent_valid = create(:tool_renting, renter: 'Adrian', tool_id: tool.id)
+
+      rent = build(:tool_renting, tool_id: tool.id)
+      rent.valid?.should eq(false)
+      rent.errors[:returned].should include('no more tools to rent')
+
+      # Check update rent with no frees
+      rent_valid.renter = 'Roth'
+      rent_valid.valid?.should eq(true)
+    end
+
+    it 'check total not under rented tools' do
+      tool = create(:tool, total: 1)
+      create(:tool_renting, tool: tool)
+      tool.total = 0
+      tool.valid?.should eq(false)
+      tool.errors[:total].should include('not allowed to change total below rented tools')
+    end
+  end
+
+  describe 'method free' do
+    it 'check renting' do
+      tool = create(:tool)
```

Comment:
I would prefer `create(:tool, total: 3)`  and then you do not have to change the factory.
Now you would still need to check the factory for the total amount.